Record status codes when middleware instrumentation is disabled (#2175)
newrelic/newrelic-ruby-agent: CHANGELOG.md

# New Relic Ruby Agent Release Notes

## dev

> Version <dev> allows the agent to record the http status code on a transaction when middleware instrumentation is disabled and fixes a bug in `NewRelic::Rack::AgentHooks.needed?.

Reviewer: ```suggestion
Version <dev> allows the agent to record the http status code on a transaction when middleware instrumentation is disabled and fixes a bug in `NewRelic::Rack::AgentHooks.needed?`.
```

> Version <dev> allows the agent to record the http status code on a transaction when middleware instrumentation is disabled and fixes a bug in `NewRelic::Rack::AgentHooks.needed?`.

Reviewer 2: Since we're now including content type...
```suggestion
Version <dev> allows the agent to record additional response information on a transaction when middleware instrumentation is disabled and fixes a bug in `NewRelic::Rack::AgentHooks.needed?`.
```

Author: Oh good catch I hadn't thought about the summary, thank you!

> - **Feature: Transactions now report http status codes when middleware instrumentation is disabled**

Reviewer 3: What about using the imperative case: `Report transaction HTTP status codes when middleware instrumentation is disabled`

> - **Feature: Transactions now report http status codes when middleware instrumentation is disabled**

Reviewer 2: Ooh, I like that! Good imperative rewrite 😄

> - **Feature: Transactions now report http status codes when middleware instrumentation is disabled**

Author: I like that, it sound nice. thank you! ad97ce59cf4f0d8cd1e3bcd2f2c46eaafb6587e2

>   Previously, if `disable_middleware_instrumentation` is set to true, the agent would not record the value of the response code on the transaction. This was due to the possibility that a middleware could be used that might alter the response, which would not be captured by the agent if the middleware instrumentation is disabled. However, based on customer feedback, the agent will now report the http status code on a transaction still when middleware instrumentation is disabled. [PR#2175](https://github.com/newrelic/newrelic-ruby-agent/pull/2175)

Reviewer: ```suggestion
  Previously, if `disable_middleware_instrumentation` is set to `true`, the agent would not record the value of the response code on the transaction. This was due to the possibility that a middleware could be used that might alter the response, which would not be captured by the agent if the middleware instrumentation is disabled. However, based on customer feedback, the agent will now report the http status code on a transaction still when middleware instrumentation is disabled. [PR#2175](https://github.com/newrelic/newrelic-ruby-agent/pull/2175)
```

>   Previously, if `disable_middleware_instrumentation` is set to `true`, the agent would not record the value of the response code on the transaction. This was due to the possibility that a middleware could be used that might alter the response, which would not be captured by the agent if the middleware instrumentation is disabled. However, based on customer feedback, the agent will now report the http status code on a transaction still when middleware instrumentation is disabled. [PR#2175](https://github.com/newrelic/newrelic-ruby-agent/pull/2175)

Reviewer 2: ```suggestion
Previously, when `disable_middleware_instrumentation` was set to `true`, the agent would not record the value of the response code on the transaction. This was due to the possibility that a middleware could alter the response, which would not be captured by the agent when the middleware instrumentation was disabled. However, based on customer feedback, the agent will now report the HTTP status code on a transaction when middleware instrumentation is disabled. [PR#2175](https://github.com/newrelic/newrelic-ruby-agent/pull/2175)
```

> Previously, when `disable_middleware_instrumentation` was set to `true`, the agent would not record the value of the response code or content type on the transaction. This was due to the possibility that a middleware could alter the response, which would not be captured by the agent when the middleware instrumentation was disabled. However, based on customer feedback, the agent will now report the HTTP status code and content type on a transaction when middleware instrumentation is disabled. [PR#2175](https://github.com/newrelic/newrelic-ruby-agent/pull/2175)

Reviewer 2: I think I might've removed the leading spaces by accident in my earlier recommendation. We usually have them, so here they are again.

```suggestion
  Previously, when `disable_middleware_instrumentation` was set to `true`, the agent would not record the value of the response code or content type on the transaction. This was due to the possibility that a middleware could alter the response, which would not be captured by the agent when the middleware instrumentation was disabled. However, based on customer feedback, the agent will now report the HTTP status code and content type on a transaction when middleware instrumentation is disabled. [PR#2175](https://github.com/newrelic/newrelic-ruby-agent/pull/2175)
```

> - **Bugfix: Resolve inverted logic of `NewRelic::Rack::AgentHooks.needed?`**

Author: That makes sense, I'm totally fine with removing them. I didn't realize that was a thing (or i forgot oops), so thank you for letting me know!

>   Previously, `NewRelic::Rack::AgentHooks.needed?` was incorrectly using inverted logic. This has now been resolved. [PR#2175](https://github.com/newrelic/newrelic-ruby-agent/pull/2175)

Author: I can do that! ad97ce59cf4f0d8cd1e3bcd2f2c46eaafb6587e2
